test/intest10.F90

```diff
@@ -0,0 +1,91 @@
+! This is a test for NCEPLIBS-bufr.
+!
+! Reads test file 'testfiles/IN_10' to test ERRWRT branches in ARALLOCF, UFBMEM, UFBMEX, and OPENBT.
+!
+! J. Ator, 3/13/2023
+
+module Share_errstr
+  ! This module is needed in order to share information between the test program and subroutine errwrt, because
+  ! the latter is not called by the former but rather is called directly from within the BUFRLIB software.
+  
+  character*4000 errstr
+  
+  integer errstr_len
+end module Share_errstr
+
+subroutine errwrt(str)
+  ! This subroutine supersedes the subroutine of the same name within the BUFRLIB software, so that we can
+  ! easily test the generation of error messages from within the library.
+  
+  use Share_errstr
+  
+  character*(*) str
+  
+  integer str_len
+  
+  str_len = len(str)
+  errstr ( errstr_len + 1 : errstr_len + str_len + 1 ) = str
+  errstr_len = errstr_len + str_len
+  
+  return
+end subroutine errwrt
+
+program intest10
+  use Share_errstr
+
+  implicit none
+
+  integer*4 isetprm
+
+  integer icnt, iunt, imesg(150), idate, iret
+
+  character cmgtag*8
+
+  print *, 'Testing reading IN_10 to test ERRWRT branches in ARALLOCF, UFBMEM, UFBMEX, and OPENBT'
+
+#ifdef KIND_8
+  call setim8b ( .true. )
+#endif
+
+  if ( ( isetprm ( 'MAXMSG', 125 ) .ne. 0 ) .or. ( isetprm ( 'MAXMEM', 125000 ) .ne. 0 ) ) stop 1
+
+  ! Test some various out-of-bounds verbosity settings, and test the errwrt branch in arallocf.
+  errstr_len = 0
+  call openbf ( 21, 'QUIET', 3 )
+  if ( index( errstr(1:errstr_len), 'ARRAYS WILL BE DYNAMICALLY ALLOCATED USING THE FOLLOWING VALUES' ) .eq. 0 ) stop 2
+  call openbf ( 21, 'QUIET', -2 )
```

Comment:
Do these calls the openbf() succeed? Seems like you are not checking anything here...

Reply:
Can you please put this explanation into the code as comments? It's far from obvious!

Reply:
OK, I'll try to expand the existing comment to make it clearer.